### PR TITLE
Extend dev styleguide with vue.js part

### PR DESCRIPTION
**Pull request by amureki**
There is a bit of refactoring, but the main change is a new vue.md styleguide:
styleguide/vue.md

### CODE REVIEW

**Comment by amureki** [comment]
@voiio/software here is a proposal on how we mount vue.js components. A real-world example is here:
https://github.com/voiio/voiio-platform/pull/5598

I know that @mostafa-anm is doing similar work with analytics - to not rely on ids, but rather introduce separate data selectors, so we know where those are used.

Please review it at your will.

File: styleguide/vue.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Vue.js
+
+## Rule zero - component mounting
+
+Use `data-vue-id` selector to mount components.
+
+### Why
+
+`id` selectors are generic and could be used for multiple purposes, which then easy to forget.
```
Comment (fidelitas) [inline styleguide/vue.md:9]:
```suggestion
`id` selectors are generic and could be used for multiple purposes. They are easy to forget.
```

File: styleguide/vue.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+Example: there was a component with certain id, which was later refactored (thus id changed).
```
Comment (fidelitas) [inline styleguide/vue.md:10]:
```suggestion
Example: There is a component with a certain id. It was later refactored, the id changed..
```

File: styleguide/vue.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+However, this id was also used for analytics, which was forgotten and not updated.
```
Comment (fidelitas) [inline styleguide/vue.md:11]:
```suggestion
However, this id was also used for analytics. It was forgotten and not updated.
```

File: styleguide/vue.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+By using data selectors, we can separate concerns and avoid issues like this.
```
Comment (fidelitas) [inline styleguide/vue.md:12]:
```suggestion

By using data selectors, we can separate concerns and avoid issues like this.
```

**Review by fidelitas — APPROVED** [review]
I had some formatting / spelling comments.
I don't know about "id" or "vue-data-ids" but I can fully understand and do support the idea.
Thanks!

**Review by SebastianKapunkt — APPROVED** [review]

**Review by mostafa-anm — APPROVED** [review]

**Pull request merged**